README.md

```diff
@@ -1,2 +1,119 @@
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`.
+
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+
+```console
+helm install my-vpn -f values.yaml ./as-helm-chart
+```
+
+## Access server custom configuration
+
+Custom configuration is possible via Kubernetes [postStart](https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/) feature.
+
+***Important: If you plan to use `sacli` in the `postStart` script, ensure that the Access Server service is fully up and running beforehand.***  
+You can find an example of how to wait for the service in the Inline Script below.
+
+If enabled, this chart runs default `scripts/configure.sh` from the chart.
+This is just an example which waits while AS initialization is finished.
+The default `scripts/configure.sh` can be overridden.
+
+### Override with Inline Script
+
+#### Add to values.yaml:
+
+```console
+postStart:
+  customScript: |
+    #!/bin/bash
+    # Waiting for AS service initialization
+    until /usr/local/openvpn_as/scripts/sacli status 2>/dev/null |grep -q '"api": "on"'
+    do
+        sleep 2
+    done
+    /usr/local/openvpn_as/scripts/sacli --user "openvpn" --new_pass "secure123" SetLocalPassword
+```
+
+#### Use External Secret
+```console
+postStart:
+  externalSecretName: my-custom-script
+```
+
+Create it:
+
+```console
+kubectl create secret generic my-custom-script \
+  --from-file=configure.sh=./custom.sh
+```
+
+## TUN device in Kubernetes
+
+A TUN device in Kubernetes is a virtual network interface that allows for packet routing between user space and the kernel, commonly used for applications like VPNs. To use a TUN device, pods typically need specific permissions, and starting from Kubernetes version 1.31.3, they must run in privileged mode to create and access the TUN device.
+
+To create and use TUN devices within Kubernetes pods, specific permissions are necessary:
+Kubernetes Versions Prior to 1.31.3: The NET_ADMIN capability was sufficient for creating TUN devices.
+Kubernetes Version 1.31.3 and Later: Pods must run in privileged mode to create TUN devices. This change was made due to security improvements that restrict access to certain system resources.
+
+Also instead of using privileged mode the Device Plugins can be used
+
+Kubernetes supports device plugins that can expose host devices to pods. For TUN devices, you can deploy a device plugin that allows access to /dev/net/tun. This can be done using a DaemonSet that runs on each node.
+
+Example Device Plugin Configuration
+Here’s a basic example of a device plugin configuration:
+
+```console
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: tun-device-plugin
+  namespace: kube-system
+spec:
+  template:
+    spec:
+      containers:
+      - name: tun-device-plugin
+        image: your-device-plugin-image
```

Comment:
Not sure whether this example is particularly useful. The actual work seems to be in creating the required code for the plugin. So

* we could check whether there is an existing OSS plugin for tun devices
* create our own
* if neither of those two, I would probably remove this example. Anyone who can write such a plugin can probably come up with the configuration on their own...